apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component, PropTypes } from 'react';
+import i18n from '@cdo/locale';
+import ProgressButton from '@cdo/apps/templates/progress/ProgressButton';
+
+export const sectionShape = PropTypes.shape({
+  id: PropTypes.number.isRequired,
+  name: PropTypes.string.isRequired,
+  loginType: PropTypes.oneOf(['word', 'email', 'picture']).isRequired,
+  stageExtras: PropTypes.bool.isRequired,
+  pairingAllowed: PropTypes.bool.isRequired,
+  numStudents: PropTypes.number.isRequired,
+  code: PropTypes.string.isRequired,
+  assignmentName: PropTypes.string,
+  assignmentPath: PropTypes.string
+});
+
+const styles = {
+  sectionName: {
+    fontSize: 18,
+    paddingTop: 12
+  },
+};
+
+export default class SectionRow extends Component {
+  static propTypes = {
+    section: sectionShape.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.onClickDelete = this.onClickDelete.bind(this);
+    this.onClickDeleteNo = this.onClickDeleteNo.bind(this);
+    this.onClickDeleteYes = this.onClickDeleteYes.bind(this);
```

Comment:
In the past someone (I think @pcardune?) suggested I use [property initializer syntax](https://babeljs.io/docs/plugins/transform-class-properties/) and arrow functions instead of self-binding in the constructor, e.g.
```jsx
export default class SectionRow extends Component {
  onClickDelete = () => {
    this.setState({deleting: true});
  };
}
```
It's clever.  Thoughts?